### ISSUE TITLE
Corrupt png files

### ISSUE THREAD

**Issue by gktval** [tensorflow/tensorflow#62406]
### Issue type
Support

### TensorFlow version
2.14

### OS platform and distribution
Windows 11

### Python version
3.9

I am getting an error: Invalid PNG data, size 49253

Part of my classification process classifying bad images from good ones. All of my png files can be opened, so I think the images are getting corrupt with they are resized. In the graphic below are a few of the "bad" images. 
![image](https://github.com/tensorflow/tensorflow/assets/7605046/bd18cefb-f001-41a3-8f57-b14379cebdc4)

After importing the images into the training dataset (code below), how I can filter out the images that became corrupt in the resizing? Right now it is throwing an error on when iterating through the normalization_layer

```
batch_size = 32
img_height = 224
img_width = 224
band_count = 4

train_ds = tf.keras.utils.image_dataset_from_directory(
  data_dir,
  validation_split=0.2,
  subset="training",
  seed=123,
  image_size=(img_height, img_width),
  batch_size=batch_size,
  color_mode="rgba")

normalization_layer = layers.Rescaling(1./255)
normalized_ds = train_ds.map(lambda x, y: (normalization_layer(x), y))
#Error here: Invalid PNG data, size 49253
image_batch, labels_batch = next(iter(normalized_ds))
```

### Standalone code to reproduce the issue

```shell
Code above
```

**Comment by gktval** [comment]
Running through all the images, I found the one causing the problem. For some reason or another it could be opened in other programs, but not in TF.
```
for imgPath in images:
    img = tf.keras.utils.load_img(
        imgPath, color_mode="rgba", target_size=(img_height, img_width)
    )

    img_array = tf.keras.utils.img_to_array(img)
```
